haskell.nix/library/.github/workflows/check.yml

```diff
@@ -0,0 +1,51 @@
+name: Nix flake check
+on: push
+
+jobs:
+  validate:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: xrefcheck
+        run: nix run github:serokell/xrefcheck
+
+      - name: reuse
+        run: nix build -L .#checks.x86_64-linux.reuse-lint
+        # Run step even if the previous one has failed
+        if: success() || failure()
+
+      - name: hlint
+        run: nix build -L .#checks.x86_64-linux.hlint
+        if: success() || failure()
+
+      - name: stylish-haskell
+        run: nix build -L .#checks.x86_64-linux.stylish-haskell
+        if: success() || failure()
+
+      - name: check-trailing-whitespace
+        run: nix build -L .#checks.x86_64-linux.trailing-whitespace
+        if: success() || failure()
+
+      # - name: shellcheck
+      #   run: nix build -L .#checks.x86_64-linux.shellcheck
+      #   if: success() || failure()
+
+  build-and-test:
+    name: ghc${{ matrix.ghc }}
+    runs-on: self-hosted
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc:
```

Comment:
Currently, it's required to hardcode the list of supported GHC versions in two places: in the CI pipeline configuration file and in `flake.nix`. It should be possible to avoid this at least for GitHub actions since it supports importing matrix from JSON files, see https://github.blog/changelog/2020-04-15-github-actions-new-workflow-features/#new-fromjson-method-in-expressions

For the rest of CI providers, I think we should at least provide a comment that this should be synced with `flake.nix`